lib/transaction/getAccountsForSimulation.ts

```diff
@@ -26,7 +26,9 @@ export async function getAccountsForSimulation(
     const { staticAccountKeys, accountKeysFromLookups } =
       tx.message.getAccountKeys({ addressLookupTableAccounts });
 
-    const staticAddresses = staticAccountKeys.map((k) => k.toString());
+    const staticAddresses = staticAccountKeys
+      .filter((k) => !k.equals(SystemProgram.programId))
+      .map((k) => k.toString());
```

Comment:
One can optimize by using a `reduce` instead, and you can avoid iterating over the array twice. You can filter and map to string in one go.